content/blog/2021/05/multicast-failure-routing/index.md

```diff
@@ -0,0 +1,276 @@
+<br>
+
+## Solution 
+
+There are many ways a user could neutralize the states set by the exception handling framework. However, for the scope of this article, we chose the following strategy. 
+- If first child route exchange never encountered an exception, then continue processing the rest of aggregation cycle as usual.
+- If the first child encountered an excetion, then introspect the incoming exchanges for success case. If found, shift the base to be the first succeess exchange and move the aggregated results on to it and continue the rest of aggregation lifecycle as usual.
+
+Updated AggregationStrategy
+```java
+public class SimpleFlowMergeAggregator implements AggregationStrategy {
+    private static final Logger LOGGER = LoggerFactory.getLogger(SimpleFlowMergeAggregator.class.getName());
+    @Override
+    public Exchange aggregate(Exchange oldExchange, Exchange newExchange) {
+        LOGGER.info("Inside aggregator " + newExchange.getIn().getBody());
+        if(oldExchange == null) {
+            String data = newExchange.getIn().getBody(String.class);
+            List<String> aggregatedDataList = new ArrayList<>();
+            aggregatedDataList.add(data);
+            newExchange.getIn().setBody(aggregatedDataList);
+            return newExchange;
+        }
+
+        if(hadException(oldExchange)) {
+            if(!hadException(newExchange)) {
+                // aggregate and swap the base
+                LOGGER.info("Found new exchange with success. swapping the base exchange");
+                List<String> oldData = oldExchange.getIn().getBody(List.class);
+                oldData.add(newExchange.getIn().getBody(String.class));
+                newExchange.getIn().setBody(oldData); // swapped the base here
+                return newExchange;
+            }
+        }
+
+        List<String> oldData = oldExchange.getIn().getBody(List.class);
+        oldData.add(newExchange.getIn().getBody(String.class));
+        oldExchange.getIn().setBody(oldData);
+
+        return oldExchange;
+    }
+
+
+    private boolean hadException(Exchange exchange) {
+
+        if(exchange.isFailed()) {
+            return true;
+        }
+
+        if(exchange.isRollbackOnly()) {
+            return true;
+        }
+
+        if(exchange.isRollbackOnlyLast()) {
+            return true;
+        }
+
+        if(((ExtendedExchange)exchange).isErrorHandlerHandledSet()
+                && ((ExtendedExchange)exchange).isErrorHandlerHandled()) {
+            return true;
+        }
+
+        return false;
+    }
+}
+````
```

Comment:
There is one extra backtick here and I can't make a suggestion for it, we need three not four here.

Reply:
Fixed this too